plugins/ssr/ssripats.ml

```diff
@@ -215,8 +215,11 @@ let rec ipat_tac1 ipat : unit tactic =
   | IPatView (clear_if_id,l) ->
       Ssrview.tclIPAT_VIEWS ~views:l ~clear_if_id
         ~conclusion:(fun ~to_clear:clr -> intro_clear clr)
-  | IPatDispatch ipatss ->
-      tclEXTEND (List.map ipat_tac ipatss) (tclUNIT ()) []
+
+  | IPatDispatch(false,[[]]) ->
+      tclUNIT ()
+  | IPatDispatch(_,ipatss) ->
```

Comment:
This seems wrong. Why doesn't the behavior depend on the value of the "strict" flag?

Reply:
Hum, I guess the "name" of the boolean is misleading.
My intention is to distinguish Dispatch (number branches = number goal) and DispatchWhenUsedAsFirstCase (number branches = number goals OR number branches = 0).

So strict means "disable the second part of the OR".

I believe the code is correct. Suggestion on how to make it more readable are welcome.

Reply:
Another point here is: if ssr first case exception than I map Case -> Dispatch, but there are scripts (old ones) where one does `case=> [] []` and the first case is a no-op, today I'd write this `case=> - []`.
This prevents compiling the first `[]` as Dispatch, since Dispatch is strict now.
I did not check how many occurrences are there. We may fix the scripts and simplify this patch.

Reply:
Can you change the name to make the code slightly more understandable. I'm afraid it looks very messy.

Reply:
Are you asking to have IPatDispatchLoose and IPatDispatchStrict ? Not sure I get what you want me to change?

Reply:
@maximedenes ping

Reply:
I meant to change the comment saying the boolean means "strict". But in fact, I think comments with what you wrote above here should make it easy enough to understand the code.